### ISSUE TITLE
CameraRig position problems when made child of a RigidBody and pad rotation enabled since Unity 2018.1.6

### ISSUE THREAD

**Issue by dwhinham** [thestonefox/VRTK#1855]
> **NOTE:** Any issue that does not follow the below template will be immediately closed and not re-opened until the template structure is adhered to.

### Precheck

 * Do not use the issues tracker for help or support (try VRTK Slack channel, Stack Overflow, etc.)
 * For proposing a new feature, please check existing open and closed issues before creating a duplicate
 * For bugs, do a quick search and make sure the bug has not yet been reported
 * Finally, be excellent to each other, and party on dudes!

### Environment

 * Source of VRTK: GitHub master
 * Version of VRTK: v3.3.0 d88c9ea
 * Version of Unity3D: 2018.1.6 upwards
 * Hardware used: Oculus Rift, but issue affects SteamVR and Simulator also
 * SDK used: Oculus Utilities 1.24.0, OVRPlugin 1.24. SteamVR and Simulator also affected.

### Steps to reproduce

1. Open the *[001 - Interactions] ControllerEvents* test scene.
2. Inside *SceneObjects*, create an empty GameObject named *Platform*.
3. Inside *Platform*, create a Cube object. Set the scale to be { 5, 0.5, 5 }, and the Y position to be 0.25 so that it rests on the ground.
4. Add a *RigidBody* component to *Platform*.
5. Drag your VR SDK's camera rig object into *Platform* so that it's now a child of the *Platform* object.
6. Set the Y position of the camera rig to be 0.25 so that it rests on top of the platform.
7. Add a new script to *Platform* called *Booster*, containing the following code:
```cs
using UnityEngine;

public class Booster : MonoBehaviour
{
    public void Boost()
    {
        GetComponent<Rigidbody>().AddRelativeForce(new Vector3(0.0f, 10.0f, 0.0f), ForceMode.Impulse);
    }
}
```
8. Add a *VRTK_ControllerEvents_UnityEvents* component to the *RightControllerScriptAlias*.
9. Create a new list entry under *On Button One Released*, drag *Platform* into the *Object* field, and set the function to be `Booster.Boost()`.
10. Create a new object under *[ExampleScene_Scripts]* called *BodyPhysics*, and add a *VRTK_BodyPhysics* component to it.
11. Uncheck *Enable Body Collisions* in the *BodyPhysics* component.
12. Add a *VRTK_TouchPadControl* component to *RightControllerScriptAlias*.
13. Add a *VRTK_RotateObjectControlAction* component to *RightControllerScriptAlias*.
14. Drag *RightControllerScriptAlias* into the *Object Control Script* field in the *VRTK_RotateObjectControlAction* component.
15. Run the game, verify rotation works. Press Button A to make the platform bounce and trigger the bug.

### Expected behavior

The player's view and position should remain fixed, relative to the platform, whilst also allowing the player to rotate their play area to look around as the platform moves.

### Current behavior

In **Unity 2018.1.5**, the expected behavior is observed.
Since **Unity 2018.1.6**, the player's relative position can change - the player can shift above or below the platform as the platform's Y position changes.

Disabling the *VRTK_RotateObjectControlAction* component is enough to prevent this from happening.

In a more complex scene where the player is able to grab and hold objects, their positions can also end up being modified and appear to fly out of the player's hands.

## Correct behavior seen in Unity 2018.1.5:
[![Correct behavior](https://thumbs.gfycat.com/CoolThinAntarcticfurseal-size_restricted.gif)](https://gfycat.com/CoolThinAntarcticfurseal)

## Incorrect behavior seen in Unity 2018.1.6+:
[![Incorrect behavior](https://thumbs.gfycat.com/KeyNaturalIberiannase-size_restricted.gif)](https://gfycat.com/KeyNaturalIberiannase)

**Comment by dwhinham** [comment]
This problem seems to have been fixed after updating to **Unity 2018.2.8f1**.

I'll close this issue for now as it looks like there was a bug with Unity itself - possibly this one: https://issuetracker.unity3d.com/issues/quaternion-dot-lookrotation-breaks-rotation-physics-when-parented-with-a-moving-object

**Comment by bddckr** [comment]
Thanks for following up on this @dwhinham!
